### PR TITLE
Improve Build workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
Please read the comments below.

### CODE REVIEW

File: .github/workflows/reusable-build.yml [hunk 1]
```diff
@@ -18,8 +18,7 @@ jobs:
     - name: Install PHP
       uses: shivammathur/setup-php@v2
       with:
-        php-version: 8.0
-        tools: composer:v2
```
Comment (szepeviktor) [inline .github/workflows/reusable-build.yml:27]: Composer v2 is the default tool.

File: .github/workflows/reusable-build.yml [hunk 2]
```diff
@@ -18,8 +18,7 @@ jobs:
...
+        php-version: '8.0'
```
Comment (szepeviktor) [inline .github/workflows/reusable-build.yml:26]: This needs to be a YAML string (not a float).

File: .github/workflows/reusable-build.yml [hunk 3]
```diff
@@ -32,31 +31,30 @@ jobs:
...
       run: npm ci
 
+    - name: Install PHP Dependencies
+      run: composer install
+
     - name: Generate autoload file
-      run: |
-        composer install
-        composer run build-vendor
+      run: composer run build-vendor
```
Comment (szepeviktor) [inline .github/workflows/reusable-build.yml:46]: One command per step.

File: .github/workflows/reusable-build.yml [hunk 4]
```diff
@@ -32,31 +31,30 @@ jobs:
...
 
     - name: Build Assets
       run: npm run build
 
     - name: Setup version
-      run: echo "VERSION=$(node -p "require('./package.json').version")" >> $GITHUB_ENV
+      run: |
+        echo "VERSION=$(jq -r '."version"' ./package.json)" >>"${GITHUB_ENV}"
```
Comment (szepeviktor) [inline .github/workflows/reusable-build.yml:54]: `jq` is included in GHA images.

File: .github/workflows/reusable-build.yml [hunk 5]
```diff
@@ -32,31 +31,30 @@ jobs:
...
 
-    - name: Tag
+    - name: Release version ${{ env.VERSION }}
       run: |
-        echo "Releasing version $VERSION ..."
```
Comment (szepeviktor) [inline .github/workflows/reusable-build.yml:57]: Put the name in `name:`.

File: .github/workflows/reusable-build.yml [hunk 6]
```diff
@@ -32,31 +31,30 @@ jobs:
...
-        git checkout -b "release-$VERSION"
-        git add -f assets/*
-        git add -f vendor/autoload.php
-        git add -f vendor/composer/*
-        git commit -m "Release $VERSION"
-        git tag "$VERSION"
+        git checkout -b "release-${VERSION}"
+        git add -f assets/ vendor/autoload.php vendor/composer/
+        git commit -m "Release ${VERSION}"
+        git tag "${VERSION}"
```
Comment (szepeviktor) [inline .github/workflows/reusable-build.yml:63]: Be on the safe side!

File: .github/workflows/reusable-build.yml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 name: Build
 on:
-  workflow_call:
+  workflow_call: null
```
Comment (szepeviktor) [inline .github/workflows/reusable-build.yml]: Nothing has a name :)

**Comment by johnbillion** [comment]
I'm going to close this off as it's quite of date now 👍 . Thanks anyway.

**Pull request closed without merging**